### ISSUE TITLE
JEP 482: bogus error on method invocation of other instance

### ISSUE THREAD

**Issue by stephan-herrmann** [eclipse-jdt/eclipse.jdt.core#2467]
```
public class Test3 extends Super {
	Test3(Test3 other) {
		other.foo(); // bogus error: 
		foo(); // error is correct
		super();
	}
	void foo() {}
}
```
Ecj reports errors on both invocations of `foo`(), but the first error ("Cannot use other.foo() in a pre-construction context") must not be raised. There is no problem in using `other`, it having the type of the enclosing class is irrelevant.

BTW, some messages say "pre-construction contest" others say "prologue". Is there a system behind this choice?

**Comment by stephan-herrmann** [comment]
> BTW, some messages say "pre-construction contest" others say "prologue". Is there a system behind this choice?

Via #2672 messages are being unified towards "in an early construction context" to reflect the new terminology from JEP 482 (yes, prologue is also used in the spec, but I believe the other terminology is fully sufficient for error messages).

**Comment by stephan-herrmann** [comment]
Resolved via #2685
